### PR TITLE
docs: Implement WalletDropdownBaseName Sub-Component

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**

Add the Base Name tab to the Wallet Component. When a user has a Base Name, this tab will direct them to their Base Name profile page. If a user does not have a Base Name, this tab will direct them to a page where they can create one!

- Add new basenames user icon.
- Update landing page wallet component to use Basenames. 
- Update the `wallet`, `Avatar`, `getAvatar`, `Name`, and `getName` docs with Basenames examples. 

| Part | Description | Status |
|---|----|---|
| 1 | [feat: Implement WalletDropdownBaseName Sub-Component #913](https://github.com/coinbase/onchainkit/pull/913) | ✅ |
| 2 | [docs: Implement WalletDropdownBaseName Sub-Component #916](https://github.com/coinbase/onchainkit/pull/916) | _In Progress_|

Video recording:
[Wallet Components & Utilities · OnchainKit - 27 July 2024 - Watch Video](https://www.loom.com/share/0a38993d78864be8a27d6d688f52f857)

Note that the `Wallet` and `Basenames` tabs are flipped in prod - Wallet is above Basenames. 
Screenshot 2024-08-12 at 2 02 14 PM

Screenshot 2024-08-12 at 2 02 01 PM

Screenshot 2024-08-12 at 2 02 42 PM

Screenshot 2024-08-12 at 2 02 55 PM

`/wallet/wallet`:
Screenshot 2024-08-12 at 2 09 13 PM

Screenshot 2024-08-12 at 2 22 03 PM

Screenshot 2024-08-12 at 2 30 40 PM

Screenshot 2024-08-12 at 2 34 36 PM

Screenshot 2024-08-12 at 2 34 43 PM

Screenshot 2024-08-12 at 2 38 35 PM

Screenshot 2024-08-12 at 4 03 15 PM

**Notes to reviewers**

**How has it been tested?**
I've done end user testing (screenshot contains old UI but the functionality is the same.):
Screenshot 2024-07-30 at 4 55 37 PM

### CODE REVIEW

**Comment by Zizzamia** [comment]
@cpcramer could you rebase from main when you have a minute? With all our tests in order, it'll help our CI pass more easily.

File: .changeset/mighty-rockets-drive.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@coinbase/onchainkit": patch
+---
+
+-**docs**: Implement WalletDropdownBaseName Sub-Component. By @cpcramer #916
```
Comment (cpcramer) [inline .changeset/mighty-rockets-drive.md:5]: todo: remove changeset from docs updates

File: src/internal/svg/baseNameSvg.tsx [hunk 2]
```diff
@@ -12,13 +12,12 @@ export const baseNameSvg = (
...
-      d="M10.99 8.01001C12.21 8.04001 13.3 8.80001 13.73 9.95001L16 16H0L2.27 9.95001C2.7 8.80001 3.79 8.03001 5.01 8.01001C5.05 8.04001 6.37 9.00001 8 9.00001C9.63 9.00001 10.95 8.04001 10.99 8.01001Z"
-      className={fill.defaultReverse}
-    />
+    <g id="Icons/User">
+      <path
+        id="Vector"
+        d="M10 20C4.47715 20 0 15.5228 0 10C0 4.47715 4.47715 0 10 0C15.5228 0 20 4.47715 20 10C20 15.5228 15.5228 20 10 20ZM12.832 6.9987C12.832 5.43461 11.5641 4.16666 10 4.16666C8.43594 4.16666 7.168 5.43461 7.168 6.9987C7.168 8.56279 8.43594 9.83073 10 9.83073C11.5641 9.83073 12.832 8.56279 12.832 6.9987ZM5.67266 11.6810L4.76738 14.0938C6.10704 15.4238 7.95093 16.2465 9.98699 16.2500L10.0130 16.2500C12.0491 16.2465 13.8930 15.4238 15.2326 14.0938L14.3273 11.6810C14.0026 10.8125 13.1794 10.2385 12.2581 10.2159C12.2279 10.2385 11.2310 10.9635 10.0000 10.9635C8.76903 10.9635 7.77215 10.2385 7.74194 10.2159C6.82059 10.2310 5.9974 10.8125 5.67266 11.6810Z"
+        className={fill.defaultReverse}
+      />
+    </g>
```
Comment (cpcramer) [inline src/internal/svg/baseNameSvg.tsx:21]: I think it's fine to include this SRC file change in this "docs" PR. 

Updating the user icon to our new image.

File: site/docs/pages/identity/avatar.mdx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { Avatar, Badge, Identity } from '@coinbase/onchainkit/identity';
 import App from '../../components/App';
+import { base } from 'viem/chains';
 
 # `<Avatar />`
 
-The `Avatar` component is used to display ENS avatar associated with Ethereum addresses.
-When an ENS avatar is not available, it defaults to blue color avatar.
+The `Avatar` component is used to display ENS or Basenames avatar associated with Ethereum addresses.
```
Comment (Zizzamia) [inline site/docs/pages/identity/avatar.mdx:7]: Can we link `Basenames` anywhere? maybe the official Basename site, so we add more context what those are.

File: site/docs/pages/identity/avatar.mdx [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
+When an avatar is not available, it defaults to blue color avatar.
```
Comment (Zizzamia) [inline site/docs/pages/identity/avatar.mdx:8]: We should mention also the default version for Basename.

File: site/docs/pages/identity/avatar.mdx (hunk 3 above)
Comment (cpcramer) [reply site/docs/pages/identity/avatar.mdx]: `Basenames` will now link to https://www.base.org/names

Screenshot 2024-08-12 at 3 04 53 PM

File: site/docs/pages/identity/get-avatar.mdx [hunk 5]
```diff
@@ -8,14 +8,20 @@ use it with Next.js or any Node.js backend.
...
 ```tsx
 import { getAvatar } from '@coinbase/onchainkit/identity';
+import { base, mainnet } from 'viem/chains';
 
-const avatar = await getAvatar({ ensName: 'vitalik.eth' });
+// For Basename
+const baseAvatar = await getAvatar({ ensName: 'vitalik.base.eth', chain: base });
+
+// For ENS
+const ensAvatar = await getAvatar({ ensName: 'vitalik.eth', chain: mainnet });
```
Comment (Zizzamia) [inline site/docs/pages/identity/get-avatar.mdx:24]: Let's split those example in two code block, and also add Return Value tab

File: site/docs/pages/wallet/wallet.mdx [hunk 6]
```diff
@@ -86,6 +87,7 @@ export function WalletComponents() {
       <WalletDropdownLink icon="wallet" href="https://wallet.coinbase.com">
         Wallet
       </WalletDropdownLink>
+      <WalletDropdownBaseName />
```
Comment (Zizzamia) [inline site/docs/pages/wallet/wallet.mdx:90]: I feel like we should document this in his own page.

File: site/docs/pages/wallet/wallet.mdx [hunk 7]
```diff
@@ -186,6 +190,44 @@ You can override component text using `text`.
   </Wallet>
 </WalletComponents>
 
+### Add Basenames support
```
Comment (Zizzamia) [inline site/docs/pages/wallet/wallet.mdx:193]: Let's have this in his own page.

File: site/docs/pages/identity/avatar.mdx (hunk 4 above)
Comment (cpcramer) [reply site/docs/pages/identity/avatar.mdx:8]: The default version for Basename is the same as ENS, therefore I've updated this to:

`Address without an ENS or Basenames avatar defaults to a plain avatar:`

File: site/docs/pages/wallet/wallet.mdx (hunk 6 above)
Comment (cpcramer) [reply site/docs/pages/wallet/wallet.mdx:90]: @Zizzamia I'm not sure if this should have its own page. It's a very simple one line component that does not take in any new custom params: 

Screenshot 2024-08-12 at 3 11 20 PM

File: site/docs/pages/identity/name.mdx [hunk 8]
```diff
@@ -1,9 +1,10 @@
 import { Badge, Name, Identity, } from '@coinbase/onchainkit/identity';
 import App from '../../components/App';
+import { base } from 'viem/chains';
 
 # `<Name />`
 
-The `Name` component is used to display ENS names associated with Ethereum addresses.
+The `Name` component is used to display ENS or Basenames associated with Ethereum addresses.
```
Comment (Zizzamia) [inline site/docs/pages/identity/name.mdx:7]: Everytime Basename is mentioned we should link to their site.

Comment (Zizzamia) [reply site/docs/pages/identity/name.mdx:7]: Or at least the first time is mentioned for every page.

File: site/docs/pages/identity/get-avatar.mdx (hunk 5 above)
Comment (cpcramer) [reply site/docs/pages/identity/get-avatar.mdx:24]: This is what the return value tab would look like:

baseEnsAvatar: https://zku9gdedgba48lmr.public.blob.vercel-storage.com/basenames/avatar/paul.base.eth/1722120524815/FWUzoZmJ_400x400-kWjr2gMvjNe9hHMs9Z9LxGVGIME3By.jpg

mainnetEnsAvatar https://euc.li/paulcramer.eth

Comment (cpcramer) [reply site/docs/pages/identity/get-avatar.mdx]: done

File: site/docs/pages/wallet/wallet.mdx (hunk 7 above)
Comment (cpcramer) [reply site/docs/pages/wallet/wallet.mdx:193]: TODO: I will fast follow with adding a separate page under `Wallet` for `WalletDropdownBaseName`, `walletDropdownLink` and `WalletDropdownDisconnect`

File: site/docs/pages/identity/get-avatar.mdx [hunk 9]
```diff
@@ -8,16 +8,44 @@ use it with Next.js or any Node.js backend.
...
+```tsx [code]
 import { getAvatar } from '@coinbase/onchainkit/identity';
+import { base, mainnet } from 'viem/chains';
+
+const baseAvatar = await getAvatar({ ensName: 'paulcramer.eth', chain: base });
+```
 
-const avatar = await getAvatar({ ensName: 'vitalik.eth' });
+```ts [return value]
+https://zku9gdedgba48lmr.public.blob.vercel-storage.com/basenames/avatar/paul.base.eth/1722120524815/FWUzoZmJ_400x400-kWjr2gMvjNe9hHMs9Z9LxGVGIME3By.jpg
```
Comment (Zizzamia) [inline site/docs/pages/identity/get-avatar.mdx:27]: Interesting link

Comment (cpcramer) [reply site/docs/pages/identity/get-avatar.mdx:27]: I know haha. It's not very pretty but the user can copy and paste that link to go directly to the basename avatar

File: site/docs/pages/identity/avatar.mdx (hunk 4 above)
Comment (cpcramer) [reply site/docs/pages/identity/avatar.mdx:8]: > When an avatar is not available, it defaults to blue color avatar.

This appears to be out of date.

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**